### DISCUSSION TITLE
Generate a schema.prisma file from a *.sql file

### DISCUSSION THREAD

**Discussion by redsuperbat** [prisma/prisma#4777] [Q&A] [answered]
Hello. 

I'm trying out prisma for the first time by migrating an existing mysql DB to prisma. I'm stuck att generating a schema.prisma file from my *.sql dump. Can anyone point me in the right direction?

**Comment by ryands17** [comment, marked as answer]
Hey @redsuperbat 👋 
You cannot directly create `schema.prisma` from a dump. You would need to connect to an existing database that has your data or import the dump that you have back in the DB.

Then, to add your tables to a `schema.prisma` file from your database is easy by using [Introspection](https://www.prisma.io/docs/concepts/components/introspection#what-does-introspection-do)

As you're using MySQL, you would need to follow this [tutorial](https://www.prisma.io/docs/getting-started/setup-prisma/add-to-existing-project-typescript-mysql) to setup Prisma with an existing database.

Let me know if it works :)

**Reply by redsuperbat** [reply]
Great! This was the answer I was looking for. Will def try out introspection. Thanks! :rocket:

**Comment by yeliex** [comment]
@ryands17 In the dev workflow of my team, we wrote DB migrations with `.sql` files.

So, in order to work with Prisma, the best way is to generate `schema.prisma` by reading SQLs.

**Reply by ryands17** [reply]
@yeliex 👋 

Yes, that's the best way to do this would be using [Introspection](https://www.prisma.io/docs/concepts/components/introspection#what-does-introspection-do) as I mentioned above and as you're using `.sql` files for migrations, Prisma supports that currently via Migrate which you can read more about [here](https://www.prisma.io/docs/concepts/components/prisma-migrate). Let me know if you have any other queries :)

**Comment by ragsav** [comment]
@ryands17 Can you please guide what are the internal functions in Prisma codebase which convert database to schema file?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
